frontend/lib/wallets/WalletConnect.ts

```diff
@@ -1,32 +1,88 @@
+import WalletConnectProvider from '@walletconnect/web3-provider';
+import { ethers } from 'ethers';
+import { formatToHexWithoutPad } from 'auctions-core/helpers/format';
+import { getChainIdByNetworkType, getNetworkTypeByChainId } from 'auctions-core/src/network';
+import { setSigner } from 'auctions-core/src/signer';
 import WalletConnectLogo from '~/assets/icons/wallets/walletconnect.svg';
 import AbstractWallet from '~/lib/wallets/AbstractWallet';
 
 export default class WalletConnect extends AbstractWallet {
     public static title = 'WalletConnect';
     public static icon = WalletConnectLogo;
-    public static downloadUrl = 'https://metamask.io/';
 
-    get isInterfaceReady() {
-        return true;
+    private static provider?: WalletConnectProvider;
+
+    private addresses?: string[];
+
+    public get address() {
+        if (Array.isArray(this.addresses) && this.addresses[0]) {
+            return this.addresses[0].toLowerCase();
+        }
+    }
+
+    async getProvider(): Promise<ethers.providers.JsonRpcProvider> {
+        const infuraId = process.env.INFURA_PROJECT_ID;
+        if (!infuraId) {
+            throw new Error(`No INFURA_PROJECT_ID env variable was provided`);
+        }
```

Comment:
Initially [you claimed](https://github.com/sidestream-tech/unified-auctions-ui/issues/353#issuecomment-1309041473) that it can work with regular RPC urls if the chainId is provided. Has this changed?

Otherwise the requirement of the issue stands to _not_ introduce any new env variables. We already have `RPC_URL` and that should be enough to provide it here either as a single url or as multiply urls derived from it (see `getDefaultNetworkConfigs`).

To be clear, this feature should also _not_ introduce new requirement to have `PRC_URL` provider to be infura. It should work with any